internal/ceb/ceb.go

```diff
@@ -42,6 +43,10 @@ type CEB struct {
 	execIdx      int64
 
 	cleanupFunc func()
+
+	mu             sync.Mutex
```

Comment:
can we constrain this to `urlAgentMu`? As an outsider reading this I'd be confused as to when `mu` needs to be held. Since we're only using it for the URL agent let's constrain it there.

Reply:
Yup! no prob.